### ISSUE TITLE
Is there a way to exclude train interactions in predict method?

### ISSUE THREAD

**Issue by Ivanclj** [lyst/lightfm#336]
Hi,

I am trying to print out the results of predictions for test data but I want to exclude train interactions in the prediction results. I noticed that predict_rank is doing the job but it seems like it only rank the non-zero value of the test interaction matrix and thus is not a full list of recommendation rank (from 0 to n_items -1 but rather a bunch of zeros and a few disjoint numbers (eg. 3,51,177)). So I could not print out the first k recommendations.  I also tried removing item_ids that has interacted with, say user 0, in train and feed the rest of item_ids into the predict method so that it will not recommend items that user 0 interacted with in train. However, the results did not match the precision at K score (set train_interactions=train). So I was wondering what should I do to solve this problem? Thanks for your help!

**Comment by maciejkula** [comment]
How about using the `predict` method?

**Comment by Ivanclj** [comment]
I tried to use the predict method. But first it does not have an argument for passing in train_interaction. Then I tried : predict(user_ids=0, item_ids=item_ids) where item_ids are remaining item_ids after removing all item_ids that have interacted with user 0 in train. However, the results did not match with the precision at K score from the evaluation module. So I was wondering what should I do here.
